compiler/ir/ir.psi2ir/src/org/jetbrains/kotlin/psi2ir/intermediate/CallBuilder.kt

```diff
@@ -84,11 +84,11 @@ val CallBuilder.explicitReceiverType: KotlinType?
 fun CallBuilder.setExplicitReceiverValue(explicitReceiverValue: IntermediateValue) {
     val previousCallReceiver = callReceiver
     callReceiver = object : CallReceiver {
-        override fun call(withDispatchAndExtensionReceivers: (IntermediateValue?, IntermediateValue?) -> IrExpression): IrExpression {
-            return previousCallReceiver.call { dispatchReceiverValue, _ ->
+        override fun call(withDispatchAndExtensionAndContextReceivers: (IntermediateValue?, IntermediateValue?, List<IntermediateValue>) -> IrExpression): IrExpression {
+            return previousCallReceiver.call { dispatchReceiverValue, _, contextReceiverValues ->
                 val newDispatchReceiverValue = if (hasExtensionReceiver) dispatchReceiverValue else explicitReceiverValue
                 val newExtensionReceiverValue = if (hasExtensionReceiver) explicitReceiverValue else null
-                withDispatchAndExtensionReceivers(newDispatchReceiverValue, newExtensionReceiverValue)
+                withDispatchAndExtensionAndContextReceivers(newDispatchReceiverValue, newExtensionReceiverValue, contextReceiverValues)
```

Comment:
If there are no PSI2IR tests where context receivers play some role in desugaring, please, add such tests.

Here I'm mostly interested in:
(1) prefix/postfix increment/decrement (`inc` and `dec` operator funs with context receivers);
(2) compound assignment (`plus` / `plusAssign`, `minus` / `minusAssign`, `times` / `timesAssign`, `div` / `divAssign` operator funs with context receivers);
(3) array access (`get` / `set` operator funs with context receivers);
(4) array access in increment/decrement and compound assignment (`get` / `set` operator funs with context receivers, combined with (1) and (2));
(5) delegated properties (`getValue` / `setValue` / `provideDelegate` operator funs with context receivers);
(6) `for` loops (`iterator` operator fun with context receivers).

Such tests would check that context receivers are propagated properly in desugaring. So far it looks like everything should work, but it's quite possible I've missed something.

Reply:
Added both irText and irBlackBox for every case you mentioned. [See the commit.](https://github.com/JetBrains/kotlin/pull/4488/commits/f8d2def9a4c6aabf8e8162246dd6500a724a42ef)